experimental/dds/tree2/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -316,132 +324,111 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
+
+		const overDstToSrc = new ChildChangeMap<RegisterId>();
+		const overSrcToDst = new ChildChangeMap<RegisterId>();
+		for (const [src, dst] of overChange.moves) {
+			const srcTagged = withIntention(src);
+			const dstTagged = withIntention(dst);
+			overSrcToDst.set(srcTagged, dstTagged);
+			overDstToSrc.set(dstTagged, srcTagged);
+		}
 
-				if (over.fieldChange.newContent !== undefined) {
-					const overContent = over.fieldChange.newContent;
-					restoredRollbackChangeId = {
-						revision: overIntention,
-						localId: over.fieldChange.id,
-					};
-					if ("revert" in overContent) {
-						restoredUndoChangeId = overContent.revert;
-					}
-				}
+		const renamedDsts = new ChildChangeMap<RegisterId>();
+		for (const [_, dst] of moves) {
+			renamedDsts.set(dst, dst);
+		}
+		for (const [src, dst] of overSrcToDst.entries()) {
+			if (!renamedDsts.has(src)) {
+				renamedDsts.set(src, dst);
 			}
+		}
 
-			for (const [id, childChange] of change.childChanges) {
-				if (id === "self") {
-					const overChildChange = overChildChanges.get(id);
-					if (over.fieldChange !== undefined) {
-						// `childChange` refers to the node existing in this field before rebasing, but
-						// that node was removed by `over`.
-						const rebasedChild = rebaseChild(
-							childChange,
-							overChildChange,
-							NodeExistenceState.Dead,
-						);
-						if (rebasedChild !== undefined) {
-							perChildChanges.set(
-								{
-									// TODO: Document this choice. This isn't really the revision that deleted the node, but
-									// the one that puts it back such that if we later ressurect it, the child changes will
-									// apply to it... this matches what the previous code/format did, but it's not well-documented
-									// why it's the right choice.
-									// See the "can rebase a node replacement and a dependent edit to the new node" test case.
-									// This might be making assumptions on sandwich rebasing a la rollback tags (which could be
-									// an obstacle for postbase)
-									revision: getIntention(
-										over.fieldChange?.revision ?? overTagged.revision,
-										revisionMetadata,
-									),
-									localId: over.fieldChange.id,
-								},
-								rebasedChild,
-							);
-						}
-					} else {
-						// `over` didn't remove the node (its fieldChange is undefined)
-						const rebasedChild = rebaseChild(
-							childChange,
-							overChildChange,
-							NodeExistenceState.Alive,
-						);
-						if (rebasedChild !== undefined) {
-							perChildChanges.set(id, rebasedChild);
-						}
-					}
+		let reservedDetachId: RegisterId | undefined = change.reservedDetachId;
+		const changeDstToSrc = new ChildChangeMap<RegisterId>();
+		const changeSrcToDst = new ChildChangeMap<RegisterId>();
+		for (const [src, dst] of moves) {
+			changeSrcToDst.set(src, dst);
+			changeDstToSrc.set(dst, src);
+		}
+
+		for (const [src, dst, target] of moves) {
+			if (target === "cellTargeting") {
```

Comment:
The code in this if block is assuming that the only source register that is being moved from with cell targeting is "self", right?
If so, an assert or comment would be nice.